src/server.py

```diff
@@ -3,39 +3,121 @@
+
+table = initdb_blueprint.table
+
+# ----------------------------------------------------------------------------
+# AUTHENTICATION
+# ----------------------------------------------------------------------------
+
+app.session_interface = InMemorySessionInterface()
+app.config.OAUTH_PROVIDER = config('OAUTH_PROVIDER')
+app.config.OAUTH_REDIRECT_URI = config('OAUTH_REDIRECT_URI')
+app.config.OAUTH_SCOPE = config('OAUTH_SCOPE')
+app.config.OAUTH_CLIENT_ID = config('OAUTH_CLIENT_ID')
+app.config.OAUTH_CLIENT_SECRET = config('OAUTH_CLIENT_SECRET')
+
+
+@app.middleware('request')
+async def add_session_to_request(request):
+    await request.app.session_interface.open(request)
+
+
+@app.middleware('response')
+async def save_session(request, response):
+    '''
+    See SESSION_ERROR in Documentation.
+    '''
+    try:
+        user_info = request['session']['user_info']
+        try:
+            request['session']['user_info'] = dict(
+                [(attr, getattr(user_info, attr))
+                  for attr in user_info.default_attrs]
+            )
+        except AttributeError:
+            pass
+    except KeyError:
+        pass
+
+    await request.app.session_interface.save(request, response)
+
+
+@app.route('/profile')
+@login_required
+async def user_profile(request, user):
+    data = 'User: {}'.format(user.email)
+    return response.text(data)
 
+# ----------------------------------------------------------------------------
+# MAIN ROUTES
+# ----------------------------------------------------------------------------
 
-@app.route('/api/links', methods=['GET'])
-async def get_links(request):
+@app.route('/', methods=['GET'])
+async def get_active_links(request):
     try:
         async with app.engine.acquire() as conn:
-            data = []
-            queryset = await conn.execute(table.select())
+            data = ''
+            queryset = await conn.execute(
+                table.select().where(
+                    table.columns['is_active'] == True
+                )
+            )
             for row in await queryset.fetchall():
-                data.append({
-                    'endpoint': row.endpoint,
-                    'url': row.url
-                })
-            return json(dumps(data), status=200)
+                data += 'Owner: {}\nEndpoint: {} \nURL: {} \n\n'.format(
+                    row.owner, row.endpoint, row.url
+                )
+            return text(data, status=200)
 
     except Exception as error:
         print(error)
         return json({'message': 'getting links failed'}, status=500)
 
 
-@app.route('/<link_endpoint>')
+@app.route('/my_links', methods=['GET'])
+@login_required
+async def owner_specific_links(request, user):
+    try:
+        async with app.engine.acquire() as conn:
+            data = 'User: {}\n\n'.format(user.email)
+            queryset = await conn.execute(
+                table.select().where(
+                    table.columns['owner'] == user.email
```

Comment:
Here, they say that the user email might not be unique for a user: https://developers.google.com/identity/protocols/oauth2/openid-connect#obtainuserinfo I guess that a user is allowed to change their email or even to delete their account and another person can create a new account with the same email address.

So I guess it's better to use `user.id` which is actually set from the `sub` claim (https://gitlab.com/SirEdvin/sanic-oauth/-/blob/master/sanic_oauth/providers.py#L36). Google says that `sub` is:

> An identifier for the user, unique among all Google accounts and never reused. A Google account can have multiple email addresses at different points in time, but the sub value is never changed. Use sub within your application as the unique-identifier key for the user. Maximum length of 255 case-sensitive ASCII characters.

https://developers.google.com/identity/protocols/oauth2/openid-connect#obtainuserinfo

Reply:
Changed lookup field from email to user id.